### PR TITLE
fix: extract protectedRoutes and adjust checking mechanism and add ne…

### PR DESCRIPTION
**Pull request by mcstover**
…w routes

This makes it easier to set and manage routes that should have the protected cache-control headers applied and adds a couple new routes (auto-deposit, join-team and instant lending routes)

### CODE REVIEW

**Review by christian14b — APPROVED** [review]
Nice

File: server/vue-middleware.js [hunk 1]
```diff
@@ -74,8 +75,11 @@ module.exports = function createMiddleware({
 		// set html response headers
 		res.setHeader('Content-Type', 'text/html');
 		// Set strict cache-control headers for protected pages
-		// eslint-disable-next-line max-len
-		if (req?.url?.match(/(checkout|settings|portfolio|lend\/saved-search|monthlygood\/thanks|process-browser-auth|register|start-verification|confirm-instant-donation|instant-donation-thanks)/g)?.length) {
+		if (req?.url && protectedRoutes.filter(route => {
+			return req?.url?.indexOf(route) !== -1;
+		}).length) {
+			// eslint-disable-next-line max-len
```
Comment (dyersituations) [inline server/vue-middleware.js:81]: Remove commented out code?

Comment (mcstover) [reply server/vue-middleware.js:81]: good catch, cleanup incoming

**Review by dyersituations — APPROVED** [review]

**Review by emuvente — APPROVED** [review]
Great!

**Pull request merged**